frontend/src/app/features/details/parcelDescriptions/parcelDescriptions.tsx

```diff
@@ -64,61 +79,74 @@ const ParcelDescriptions = () => {
     graphQLPropertyName: any,
     newSortByInputValue: string | [Date, Date],
   ) => {
-    dispatch(
-      setSortByInputValue({
-        ...sortByInputValue,
-        [graphQLPropertyName]: newSortByInputValue,
-      }),
-    );
+    sortByInputValue = {
+      ...sortByInputValue,
+      [graphQLPropertyName]: newSortByInputValue,
+    };
     switch (newSortByInputValue) {
       case 'newToOld':
-        dispatch(setSortBy('date_noted'));
-        dispatch(setSortByDir('DESC'));
+        sortBy = 'date_noted';
+        sortByDir = 'DESC';
         break;
       case 'oldTonew':
-        dispatch(setSortBy('date_noted'));
-        dispatch(setSortByDir('ASC'));
+        sortBy = 'date_noted';
+        sortByDir = 'ASC';
         break;
       default:
-        dispatch(setSortBy('id'));
-        dispatch(setSortByDir('ASC'));
+        sortBy = 'id';
+        sortByDir = 'ASC';
         break;
     }
+    needsUpdate = true;
+    fetchNewParcelDescriptions();
   };
 
   const handleTableSortChange = (column: TableColumn, descending: boolean) => {
     switch (column.graphQLPropertyName) {
       case 'descriptionType':
-        dispatch(setSortBy('description_type'));
+        sortBy = 'description_type';
         break;
       case 'idPinNumber':
-        dispatch(setSortBy('id_pin_number'));
+        sortBy = 'id_pin_number';
         break;
       case 'dateNoted':
-        dispatch(setSortBy('date_noted'));
+        sortBy = 'date_noted';
         break;
       case 'landDescription':
-        dispatch(setSortBy('land_description'));
+        sortBy = 'land_description';
         break;
       default:
-        dispatch(setSortBy('id'));
+        sortBy = 'id';
         break;
     }
-    dispatch(setSortByDir(descending ? 'DESC' : 'ASC'));
+    sortByDir = descending ? 'DESC' : 'ASC';
+    needsUpdate = true;
+    fetchNewParcelDescriptions();
   };
 
-  useEffect(() => {
-    const params: IFetchParcelDescriptionParams = {
-      siteId: siteId,
-      page: currentPage,
-      pageSize: resultsPerPage,
+  const fetchNewParcelDescriptions = () => {
+    if (needsUpdate) {
+      requestStatus = RequestStatus.loading;
```

Comment:
Similar to my above comment, this should maintained in slice and it should be dispatched as an action.